libp2p/6-interfaces.md

```diff
@@ -1,139 +1,99 @@
 
-~~The network is abstracted through the swarm which presents a simplified interface for the remaining layers to have access to the network. This interface should look like:~~
+A Peer Routing service offers a way for libp2p Node to find the PeerInfo of another Node, so that it can dial to that node. In it is most pure form, a Peer Routing module should have a interface that given a 'key', a set of PeerInfos are returned.
+See https://github.com/diasdavid/abstract-peer-routing for the interface and tests.
 
-- `sw.addTransport(transport, [options, dialOptions, listenOptions])` - Add a transport to be supported by this swarm instance. Swarm expects it to implement the [abstract-transport](https://github.com/diasdavid/abstract-transport) interface.
-- `sw.addUpgrade(connUpgrade, [options])` - A connection upgrade must be able to receive and return something that implements the [abstract-connection](https://github.com/diasdavid/abstract-connection) interface.
-- `sw.addStreamMuxer(streamMuxer, [options])` - Upgrading a connection to use a stream muxer is still considered an upgrade, but a special case since once this connection is applied, the returned obj will implement the abstract-stream-muxer interface.
-- `sw.dial(PeerInfo, options, protocol, callback)` - PeerInfo should contain the ID of the peer and its respective multiaddrs known.
-- `sw.handleProtocol(protocol, handlerFunction)` - enable a protocol to be registered, so that another peer can open a stream to talk with us to that specific protocol
+## 6.3 Swarm
 
-The following figure represents how the network level pieces, are tied together:
+Current interface available and updated at:
 
-```
-┌ ─ ─ ─ ─ ┌ ─ ─ ─ ─ ┌ ─ ─ ─ ─ ┌───────────┐
- mounted │ mounted │ mounted ││Identify   │
-│protocol │protocol │protocol │(mounted   │
- 1       │ 2       │ ...     ││ protocol) │
-└ ─ ─ ─ ─ └ ─ ─ ─ ─ └ ─ ─ ─ ─ └───────────┘
-┌─────────────────────────────────────────┐
-│             swarm                       │
-└─────────────────────────────────────────┘
-┌─────────────────────────────────────────┐
-│            connection                   │
-└─────────────────────────────────────────┘
-┌───────────────┐┌───────────┐┌───────────┐
-│Transport      ││multistream││ stream    │
-│(TCP, UDP, etc)││           ││ muxer     │
-└───────────────┘└───────────┘│┌ ─ ─ ─ ─ ┐│
-                              │  spdy     │
-                              │└ ─ ─ ─ ─ ┘│
-                              │┌ ─ ─ ─ ─ ┐│
-                              │ multiplex │
-                              │└ ─ ─ ─ ─ ┘│
-                              │┌ ─ ─ ─ ─ ┐│
-                              │ QUIC      │
-                              │└ ─ ─ ─ ─ ┘│
-                              │┌ ─ ─ ─ ─ ┐│
-                              │ others    │
-                              │└ ─ ─ ─ ─ ┘│
-                              └───────────┘
-```
+https://github.com/diasdavid/js-libp2p-swarm#usage
```

Comment:
It is currently available on the swarm README, I can copy over. I was hoping for the bikeshed to happen there, so that we don't have to maintain two copies while we discuss the API